src/libratbag-hidraw.c

```diff
@@ -1660,3 +1660,49 @@ ratbag_hidraw_read_input_report_index(struct ratbag_device *device, uint8_t *buf
 
 	return rc >= 0 ? rc : -errno;
 }
+
+int
+ratbag_hidraw_read_input_report_filter(struct ratbag_device *device, uint8_t *buf, size_t len, int *(*filter)(uint8_t *buf, size_t len))
```

Comment:
style nitpick: having the function pointer on the same line wraps it in github (and probably in most narrower terminals) so it makes it slightly confusing. Also, any reason you don't use a `typedef` for the function signature?